lib/publiccloud/provider.pm

```diff
@@ -705,6 +705,23 @@ sub terraform_param_tags
     return encode_json($tags);
 }
 
+=head2 terraform_query
+
+Query the terraform data structure.
+Input: jq-query string; empty = default query; '' = full json structure.
+
+=cut
+
+sub terraform_query {
+    my ($self, $qry) = @_;
+    # default query
+    $qry //= ".vm_name.value[0]";
+    assert_script_run('cd ~/terraform');
+    $qry = script_output("terraform output -json | jq -r \"$qry\" 2>/dev/null", proceed_on_failure => 1);
```

Comment:
I don't think reusing `query` for return value is good idea 
    
```suggestion
    my $ret = script_output("terraform output -json | jq -r \"$qry\" 2>/dev/null", proceed_on_failure => 1);
```

Reply:
done